docs/topics/signals.txt

```diff
@@ -190,6 +190,14 @@ In some circumstances, the code connecting receivers to signals may run
 multiple times. This can cause your receiver function to be registered more
 than once, and thus called multiple times for a single signal event.
 
+Although :meth:`~django.apps.AppConfig.ready` is designed to be called
```

Comment:
I think we should move this note after: _"In some circumstances, the code connecting receivers to signals may runmultiple times. "_, e.g.
```diff
diff --git a/docs/topics/signals.txt b/docs/topics/signals.txt
index 30c8299fae..261027932c 100644
--- a/docs/topics/signals.txt
+++ b/docs/topics/signals.txt
@@ -187,8 +187,12 @@ Preventing duplicate signals
 ----------------------------
 
 In some circumstances, the code connecting receivers to signals may run
-multiple times. This can cause your receiver function to be registered more
-than once, and thus called multiple times for a single signal event.
+multiple times. For example, the :meth:`~django.apps.AppConfig.ready` method
+may be executed more than once during testing, another cause is if you import
+the module where you define signals from somewhere in your project (the signal
+registration will run as many times as it is imported). This can cause your
+receiver function to be registered more than once, and thus called multiple
+times for a single signal event.
 
 If this behavior is problematic (such as when using signals to
 send an email whenever a model is saved), pass a unique identifier as
```

Reply:
I gave it a try. I'm finding it's getting challenging to decide exactly how to express and describe this (gone through three or four drafts), but I agree in the majority of what you've recommended. Resolving for now, since I've changed the wording. Please review the newest iteration (there is some duplication of wording/concept, but I'm unsure that it's a bad thing).